src/tool/cpp/cppwinrt/strings/base_implements.h

```diff
@@ -1284,6 +1291,44 @@ namespace winrt
         }
 
     private:
+
+#ifdef _DEBUG
+        bool is_stack_address(void const* const address) noexcept
+        {
+            struct memory_basic_information
+            {
+                void* BaseAddress;
+                void* AllocationBase;
+                uint32_t AllocationProtect;
+                size_t RegionSize;
+                uint32_t State;
+                uint32_t Protect;
+                uint32_t Type;
+            };
+
+            static constexpr uint32_t page_guard{ 0x100 };
+
+            memory_basic_information info{};
+            WINRT_VirtualQuery(address, &info, sizeof(info));
```

Comment:
One thing I omitted from my proof of concept:  You should check the return value of VirtualQuery and make sure that it is equal to sizeof(info) before using the info structure.

Reply:
👍